File: python/examples/advanced_sdk.py
# Mycelium — advanced SDK usage with manual prime/record
# python python/examples/advanced_sdk.py

from dotenv import load_dotenv
load_dotenv()

from mycelium import prime, record, build_goal, RunOutcome


def call_my_agent(domain: str, goal: str) -> dict:
    # Replace with your own TinyFish integration
    return {
        "ok": True,
        "steps": ["navigated", "clicked", "extracted"],
        "errors": [],
        "raw_text": f"Successfully extracted data from {domain}",
        "data": {"result": "example"},
    }


def my_pipeline(domain: str, goal: str):
    primed = prime(domain, goal)
    print(f"loaded {primed.hints_loaded} hints for {domain}")

    enriched_goal = build_goal(goal, primed)

    agent_result = call_my_agent(domain, enriched_goal)

    recorded = record(RunOutcome(
        domain=domain,
        goal=goal,
        success=agent_result["ok"],
        steps=agent_result["steps"],
        errors=agent_result["errors"],
        raw=agent_result["raw_text"],
    ))

    print(f"saved {recorded.hints_extracted} new hints ({recorded.hints_total} total)")
    return agent_result


my_pipeline("amazon.com", "find the price of AirPods Pro")

File: js/examples/advanced-sdk.ts
// Mycelium — advanced SDK usage with manual prime/record
// npx tsx examples/advanced-sdk.ts

import 'dotenv/config'
import { prime, record, buildGoal } from "../index.ts"

async function myPipeline(domain: string, goal: string) {
  const primed = await prime(domain)
  console.log(`loaded ${primed.hintsLoaded} hints for ${domain}`)

  const enrichedGoal = buildGoal(goal, primed)

  // Replace with your own TinyFish integration
  const tinyfishResult = await callMyAgent(domain, enrichedGoal)

  const recorded = await record({
    domain,
    goal,
    success: tinyfishResult.ok,
    steps: tinyfishResult.steps,
    errors: tinyfishResult.errors,
    raw: tinyfishResult.rawText,
  })

  console.log(`saved ${recorded.hintsExtracted} new hints (${recorded.hintsTotal} total)`)
  return tinyfishResult
}

async function callMyAgent(domain: string, goal: string) {
  return {
    ok: true,
    steps: ["navigated", "clicked", "extracted"],
    errors: [],
    rawText: `Successfully extracted data from ${domain}`,
    data: { result: "example" },
  }
}

myPipeline("amazon.com", "find the price of AirPods Pro")
